workers/check/time_based.py

```diff
@@ -24,6 +26,12 @@ def start():
 def callback(client_state, channel, method, properties, input_message):
     dataset_id = input_message["dataset_id"]
 
+    if not is_step_required(settings.TIME_BASED_STEP):
```

Comment:
I prefer a common `finish` function, in case finishing becomes more complicated in future.

Also, this doesn't currently `ack` the message. That needs to be included in the `finish` function.

Do the same for other workers.

Reply:
Agree. I thought the same but wasn't sure if a separate function was proper enough or not, as right now we only have two lines (3 when we add the ack), but yes, it could be more complicated in the future.

Reply:
@jpmckinney, one question: do we also need to mark the dataset state as complete, or is that not necessary when skipping the step?

Reply:
I think it's safest to do all state changes.